src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs

```diff
@@ -44,47 +45,52 @@ public static string GetDefaultTestDir()
 
         public string TestRoot { get; set; } = GetDefaultTestDir();
 
-        public void AddDir(bool numberIsFloat = false, string directory = "")
+        public void AddDir(Dictionary<string, bool> setup, string directory = "")
         {
             directory = Path.GetFullPath(directory, TestRoot);
             var allFiles = Directory.EnumerateFiles(directory);
 
-            AddFile(numberIsFloat, allFiles);
+            AddFile(setup, allFiles);
         }
 
-        public void AddFile(bool numberIsFloat, params string[] files)
+        public void AddFile(Dictionary<string, bool> setup, params string[] files)
         {
             var x = (IEnumerable<string>)files;
-            AddFile(numberIsFloat, x);
+            AddFile(setup, x);
         }
 
-        public void AddFile(bool numberIsFloat, IEnumerable<string> files)
+        public void AddFile(Dictionary<string, bool> setup, IEnumerable<string> files)
         {
             foreach (var file in files)
             {
-                AddFile(numberIsFloat, file);
+                AddFile(setup, file);
             }
         }
 
         // Directive should start with #, end in : like "#SETUP:"
         // Returns true if matched; false if not. Throws on error.
-        private static bool TryParseDirective(string line, string directive, ref string param)
-        {
+        private static bool TryParseDirective(string line, string directive, out string param)
+        {
             if (line.StartsWith(directive, StringComparison.OrdinalIgnoreCase))
-            {
-                if (param != null)
-                {
-                    throw new InvalidOperationException($"Can't have multiple {directive}");
-                }
-
-                param = line.Substring(directive.Length).Trim();
+            {
+                param = line.Substring(directive.Length).Trim();
+
+                // strip any end of line comment
+                if (param.LastIndexOf("//") >= 0)
+                {
+                    param = param.Substring(0, param.LastIndexOf("//"));

```

Comment:
>LastIndexOf("//")); [](http://example.com/codeflow?start=53&length=19)

Why not .IndexOf? With LastIndexOf we wouldn't remove all comments from `#setup: FooBar //// This is a setup`, for example

Reply:
Good call, thank you!